### DISCUSSION TITLE
How do you handle wrong GPS data points in MTB?

### DISCUSSION THREAD

**Discussion by aarejogger** [mytourbook/mytourbook#1243] [General]
Once a month my watch (Garmin Forerunner 235) records GPS data incorrectly, see attached picture. At 9km running distance an incorrect recording of 500m is noticeable, distance and pace are wrong. How do you handle this situation in MTB? Any recommendation? 
I already searched for a method to correct the distance of my tour manullly in the Tour Editor or to edit the data points in the 2D tour map. Both is not working, I guess. My first priority is to correct manually the distance in the Tour Editor. PS: I am an IT dummy - no chance to edit the database...
Thanks, Michael
![Running](https://github.com/mytourbook/mytourbook/assets/155064080/569ba098-bbc9-49e0-9a74-309754e6964f)

**Comment by wolfgang-ch** [comment]
Currently there is no tool/feature in MT to correct geo positions easily to align to a path, however the tour editor provides the possibility to edit each geo position manually.

**Comment by aarejogger** [comment]
Thanks for the answer, Wolfgang! Your MTB is full of functions, that is really great :-)
I now tried some of them: In the Tour Editor / Time Slices I deleted 347 of the 350 wrong data points with the option "Delete Time Slices and Keep Time". For the other 3 wrong data points I changed the values for longitude and latitude manually with the correct values. So I have three correct points instead of 350 wrong points - correctly shown on the map! Then I used the function "Compute distance Values from Geo Positions...". The distance is now corrected and correct, great! Unfortunately with this function (or with the delete of the 347 datapoints, which changed these parameters too) the elevation (m) of each point, the elevation gain / loss (%) and the pace (km/h) of ALL datapoints were changed. The results are far from reality now. So the good thing "distance is corrected" comes with new wrong things. 
Have I used a specific function in a wrong way? Is it possible to recalculate just the distance or to recalculate just the three datapoints? 
![Running2](https://github.com/mytourbook/mytourbook/assets/155064080/e66cad4c-332b-49eb-b4b0-6fc60327032e)

**Reply by wolfgang-ch** [reply]
>Unfortunately with this function (or with the delete of the 347 datapoints, which changed these parameters too) the elevation (m) of each point, the elevation gain / loss (%) and the pace (km/h) of ALL datapoints were changed. The results are far from reality now. So the good thing "distance is corrected" comes with new wrong things.

>Have I used a specific function in a wrong way? 

No

>Is it possible to recalculate just the distance or to recalculate just the three datapoints?

All compute algorithms will compute always all data point otherwise it would be much more complex as it is now and you have to keep the positons where to compute and where not to compute.
